pkg/crd/markers/patch_validation.go

```diff
@@ -64,3 +69,22 @@ func (m FeatureSetEnum) ApplyToSchema(schema *apiext.JSONSchemaProps) error {
 	schema.Enum = vals
 	return nil
 }
+
+type FeatureSetXValidation struct {
+	FeatureSetNames []string `marker:"featureSet"`
+	Rule            string
+	Message         string `marker:",optional"`
+}
+
+func (m FeatureSetXValidation) ApplyToSchema(schema *apiext.JSONSchemaProps) error {
+	if !RequiredFeatureSets.HasAny(m.FeatureSetNames...) {
+		return nil
+	}
+
+	validation := XValidation{
+		Rule:    m.Rule,
+		Message: m.Message,
+	}
+
+	return validation.ApplyToSchema(schema)
```

Comment:
If the condition is true, treat the validation as non-conditional one through the default `XValidation` implementation.